packages/autocomplete-preset-algolia/src/utils/getAppIdAndApiKey.ts

```diff
@@ -0,0 +1,12 @@
+import type { SearchClient } from '../types';
+
+export function getAppIdAndApiKey(
+  searchClient: SearchClient
+): { appId: string; apiKey: string } {
+  const { headers, queryParameters } = searchClient.transporter;
```

Comment:
Hi @dhayab, I am not sure if we are using autocomplete incorrectly, but we ran into an issue where the `transporter` wasn't checked and the JS code was trying to pull headers from `undefined`. Could you help with this? We have a fix in a place where we are using version 1.8.3 of `autocomplete-preset-algolia`. We could be using it wrong, I am not sure.

https://github.com/algolia/algoliasearch-shopify/pull/1520

Reply:
Hi Bryan, this is probably because you have the v3 of `algoliasearch` in your dependencies. Autocomplete relies on the v4, which is what causes the issue right now.

If you're not using the javascript search client directly, you can safely update it to the latest v4 version as InstantSearch and Autocomplete have support for it.

Reply:
Ahhh wonderful thank you!